addons/website_crm/models/website_visitor.py

```diff
@@ -0,0 +1,16 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+
+
+class WebsiteVisitor(models.Model):
+    _inherit = 'website.visitor'
+
+    lead_ids = fields.One2many('crm.lead', 'visitor_id', string='Leads')
+    lead_count = fields.Integer('# Leads', compute="_compute_lead_count")
+
+    @api.depends('lead_ids')
```

Comment:
lead_ids.visitor_id actually; changing a tag on lead should not recompute this field (ok, it is not stored, but let us give short triggers)